code/modules/mining/ore_detector.dm

```diff
@@ -0,0 +1,100 @@
+/obj/item/ore_detector
+	name = "ore detector"
+	desc = "A device capable of locating and displaying ores to the average untrained hole explorer."
+	icon = 'icons/obj/contained_items/tools/ore_scanner.dmi'
+	icon_state = "ore_scanner"
+	item_state = "ore_scanner"
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	force = 1
+	var/active = FALSE
+	var/mob/our_user
+	var/search_ore
+	var/list/detection_types
+	var/ping_rate = 4 SECONDS
+	var/last_ping = 0
+
+	var/list/ore_names
+
+/obj/item/ore_detector/examine(mob/user, distance)
+	if(..(user, 1))
+		to_chat(user, FONT_SMALL(SPAN_NOTICE("Alt-click to set the ore you wish to search for.")))
+
+/obj/item/ore_detector/Destroy()
+	deactivate()
+	return ..()
+
+/obj/item/ore_detector/update_icon()
+	icon_state = "ore_scanner[active ? "-active" : ""]"
+
+/obj/item/ore_detector/attack_self(mob/user)
+	if(!search_ore)
+		to_chat(user, SPAN_WARNING("You haven't set an ore to search for yet!"))
```

Comment:
QoL thing: maybe this should instead start the ore selection process?

Reply:
i would consider it, but i just revamped it with a vueui so you can choose to detect multiple ore types, which would make it seem weird